docs: Add workflow for authenticated registries (#559)
confidential-containers/cloud-api-adaptor: docs/registries-authentication.md

Registry authentication file can be provided either statically or at runtime.

### Understanding workflow

> - For pulling images from authenticated registry you need **attestation-agent(AA)** in the podvm. The role of AA is to provide the registry secret to kata-agent. The podvm image that you are using should be built with `AA_KBC`. This ensures that [agent-config.toml](../podvm/files/etc/agent-config.toml) in podVM should have `aa_kbc_params = "offline_fs_kbc::null"` set.

Author: Yes, I think `aa_kbc_params` this is set when we build the image with `AA_KBC="offline_fs_kbc` right? Will mention it

> - For pulling images from authenticated registry you need **attestation-agent(AA)** in the podvm. The role of AA is to provide the registry secret to kata-agent. The podvm image that you are using should be built with `AA_KBC="offline_fs_kbc`. This ensures that [agent-config.toml](../podvm/files/etc/agent-config.toml) in podVM should have `aa_kbc_params = "offline_fs_kbc::null"` set.

Reviewer: ```suggestion
- For pulling images from authenticated registries you need  the [**attestation-agent**](https://github.com/confidential-containers/attestation-agent) in the podvm. The role of the attestation-agent is to provide the registry credentials to the `kata-agent`. The podvm image that you are using should be built with `AA_KBC="offline_fs_kbc`. This ensures that [agent-config.toml](../podvm/files/etc/agent-config.toml) in podVM should have `aa_kbc_params = "offline_fs_kbc::null"` set.
```

> - The registry secret is available in a file inside the Pod VM image. The config `aa_kbc_params = "offline_fs_kbc::null` tells the attestation-agent to retrieve secrets from the **local filesystem (fs)**. The registry secret is embedded in a resources file on a fixed path on the local fs i.e `/etc/aa-offline_fs_kbc-resources.json`

Reviewer: ```suggestion
- The registry credentials also need to be available in a file inside the Pod VM image. The config `aa_kbc_params = "offline_fs_kbc::null` tells the attestation-agent to retrieve secrets from the **local filesystem**. The registry credentials are embedded in a resources file on a fixed path on the local filesystem: `/etc/aa-offline_fs_kbc-resources.json`
```

> - CAA provides the secret to the local fs in the podvm image by attaching it as a config-drive which is read by `cloud-init` and copied to `/etc/aa-offline_fs_kbc-resources.json`.

Reviewer 2: >  is read by `cloud-init` and copied to `/etc/aa-offline_fs_kbc-resources.json`.
maybe something like, "secret is converted and copied using cloud-init to `/etc/aa-offline_fs_kbc-resources.json` on the podvm"

Also as mentioned below it can either be provided using caa or it can be embedded in the image (would be nice to also mention if possible)
I'm also not sure about the term "config-drive" what is it refers to?

Author: Thank you. Updated

> - CAA provides the secret to the local fs in the podvm image by attaching it. This secret is converted and copied using `cloud-init` to `/etc/aa-offline_fs_kbc-resources.json` on the podvm.

Reviewer: ```suggestion
- The cloud-api-adaptor (CAA) provides the secret to the local fs in the podvm image by attaching it. This secret is converted and copied using `cloud-init` to `/etc/aa-offline_fs_kbc-resources.json` on the podvm.
```

}
EOF
```

> - Add `AA_KBC="offline_fs_kbc"` prior to the `make image` step

Reviewer 2: The reason i added it this way is because instructions are located on different file, so i'd suggest to give context by switching "run command" with "build image with" or "make sure to build image with"
